python/pyiceberg/typedef.py

```diff
@@ -30,3 +30,4 @@ def update(self, *args: Any, **kwargs: Any) -> None:  # type: ignore
 
 Identifier = Tuple[str, ...]
 Properties = Dict[str, str]
+Configuration = Dict[str, "Configuration"]  # type: ignore
```

Comment:
Still looking for someone to implement mypy support for this: https://github.com/python/mypy/issues/731

Reply:
Should this give an option for strings as leaf nodes?

```python
Configuration = Dict[str, str | "Configuration"]
```

Reply:
Ah, great catch! 👍🏻